src/ImmutableX.ts

```diff
@@ -636,19 +636,19 @@ export class ImmutableX {
 
   /**
    * Completes a Withdrawal
-   * @param ethSigner - the L1 signer
+   * @param walletConnection - the pair of L1/L2 signers
    * @param starkPublicKey - the Signer address
    * @param token - the token
    * @returns a promise that resolves with the transaction
    * @throws {@link index.IMXError}
    */
   public completeWithdrawal(
-    ethSigner: EthSigner,
+    walletConnection: WalletConnection,
```

Comment:
this will require change in downstream dependencies like `ts-immutable-sdk`. Not sure how much work that will entail yet...